spec/models/user_spec.rb

```diff
@@ -14,6 +14,16 @@
 
     create(:user, username: "newbie")
     expect { create(:user, username: "newbie") }.to raise_error
+
+    create(:user, username: "underscores_and-dashes")
```

Comment:
I'd prefer to test validations more explicitly, like the later tests in this file do.

But this comment does not block merging. I see how you're repeating the existing pattern so I don't want to add work unnecessarily. If you'd like to change the tests to match that style I'd appreciate it, but you can leave them alone and I'll update them.